lib/codegen/fromcto/jsonschema/jsonschemavisitor.js

```diff
@@ -411,6 +406,34 @@ class JSONSchemaVisitor {
                 jsonSchema.description = 'The instance identifier for this type';
             }
 
+        } else if (ModelUtil.isMap?.(field)) {
+            const mapDeclaration = field.getParent().getModelFile().getModelManager().getType(field.getFullyQualifiedTypeName());
+
+            let valueDeclaration = mapDeclaration.getModelFile()
```

Comment:
This makes the same assumption that the type for the map value is in the same model as the map decl.